src/templates/core/dependencies.test.ts

```diff
@@ -0,0 +1,175 @@
+import { prompt } from 'inquirer';
+
+import { remove } from '@/helpers/file';
+import { AwsOptions } from '@/templates/aws';
+import { applyCommon, applyVpc } from '@/templates/aws/addons';
+
+import { applyCore } from '.';
+import { isAWSModuleAdded, requireAWSModules } from './dependencies';
+
+jest.mock('inquirer');
+
+describe('Dependencies', () => {
+  const projectDir = 'dependencies-test';
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    remove('/', projectDir);
+  });
+
+  describe('.isAWSModuleAdded', () => {
+    describe('given valid module name', () => {
+      describe('when module is added', () => {
+        it('returns true', () => {
+          const options: AwsOptions = {
+            projectName: projectDir,
+            provider: 'aws',
+            infrastructureType: 'advanced',
+            awsRegion: 'ap-southeast-1',
+          };
+
+          applyCore(options);
+          applyVpc(options);
+
+          expect(isAWSModuleAdded('vpc', options.projectName)).toBe(true);
+        });
+      });
+
+      describe('when module is not added', () => {
+        it('returns false', () => {
+          const options: AwsOptions = {
+            projectName: projectDir,
+            provider: 'aws',
+            infrastructureType: 'advanced',
+            awsRegion: 'ap-southeast-1',
+          };
+
+          expect(isAWSModuleAdded('vpc', options.projectName)).toBe(false);
+        });
+      });
+    });
+
+    describe('given INVALID module name', () => {
+      it('throws an error', () => {
+        const options: AwsOptions = {
+          projectName: projectDir,
+          provider: 'aws',
+          infrastructureType: 'advanced',
+          awsRegion: 'ap-southeast-1',
+        };
+
+        expect(() => isAWSModuleAdded('azure', options.projectName)).toThrow(
+          'Module `azure` is not supported'
+        );
+      });
+    });
+  });
+
+  describe('.requireAWSModule', () => {
+    describe('given valid module name', () => {
+      describe('when module was added', () => {
```

Comment:
What exactly do we mean with `module was added`? 👀 
How is it different than the previous `module is added` (I see we have an additional call to the `applyVpc` function...).

Should we try to find a more explicit way to describe this test and context? 🤔